File: projetos/BreviarioMaconicoAndroid/app/src/main/java/com/renatocamargo/breviariomaconico/ReadingMetricsController.kt
package com.renatocamargo.breviariomaconico

import kotlin.math.roundToInt

internal object ReadingMetricsController {
    fun progress(completed: Int, total: Int): Float {
        if (total <= 0) return 0f
        return (completed.toFloat() / total.toFloat()).coerceIn(0f, 1f)
    }

    fun percentage(completed: Int, total: Int): Int =
        (progress(completed, total) * 100).roundToInt()
}

File: BibliotecaMaconica_Dev/Models/ReadingMetricsController.swift
import Foundation

enum ReadingMetricsController {
    static func progress(completed: Int, total: Int) -> Double {
        guard total > 0 else { return 0 }
        return min(max(Double(completed) / Double(total), 0), 1)
    }

    static func percentage(completed: Int, total: Int) -> Int {
        Int((progress(completed: completed, total: total) * 100).rounded())
    }
}
